File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Event.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class Event(
  val event: String,
  val userCookieID: String,
  val url: String,
  val source: String,
  val argument: String? = null,
  val publicArgument: String? = null,
  val client: String,
  val connectedSiteID: String? = null,
  val hashedLicenseKey: String? = null,
)

File: agent/bindings/csharp/event.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class Event
  {

    [JsonPropertyName("event")]
    public string Event { get; set; }

    [JsonPropertyName("userCookieID")]
    public string UserCookieID { get; set; }

    [JsonPropertyName("url")]
    public string Url { get; set; }

    [JsonPropertyName("source")]
    public string Source { get; set; }

    [JsonPropertyName("argument")]
    public string Argument { get; set; }

    [JsonPropertyName("publicArgument")]
    public string PublicArgument { get; set; }

    [JsonPropertyName("client")]
    public string Client { get; set; }

    [JsonPropertyName("connectedSiteID")]
    public string ConnectedSiteID { get; set; }

    [JsonPropertyName("hashedLicenseKey")]
    public string HashedLicenseKey { get; set; }
  }
}
